src/ui/menu/credits/credits_popup.cpp

```diff
@@ -39,87 +42,144 @@ bool GlobedCreditsPopup::setup() {
-    ADD_PLAYER(staff, "ItzKiba", 4569963, 15447500, 110, 6, 10, 42);
-    ADD_PLAYER(staff, "NotTerma", 4706010, 16077956, 379, 98, 36, 43);
-    ADD_PLAYER(staff, "ciyaqil", 13640460, 127054504, 84, 12, 40, 40);
-    ADD_PLAYER(staff, "TheKiroshi", 6442871, 31833361, 1, 8, 11, 63);
-    ADD_PLAYER(staff, "Jiflol", 8262191, 61991331, 135, 44, 12, -1);
-    ADD_PLAYER(staff, "LimeGradient", 7214334, 42454266, 37, 2, 12, 2);
-    ADD_PLAYER(staff, "ManagerMagolor", 25450870, 195501521, 101, 43, 7, -1);
-    ADD_PLAYER(staff, "skrunkly", 14462068, 146500573, 37, 17, 19, -1);
-
-    ADD_PLAYER(staff, "Capeling", 18226543, 160504868, 141, 76, 40, 40);
-    ADD_PLAYER_NICK(staff, "Awesomeoverkill", "ninXout", 7479054, 47290058, 77, 1, 5, -1);
-    ADD_PLAYER(staff, "NikoSando", 737532, 4354472, 433, 16, 6, 12);
-
-    auto* cellStaff = Build<GlobedCreditsCell>::create("Staff", true, staff)
-        .id("cell-staff"_spr)
-        .parent(scrollLayer->m_contentLayer)
-        .collect();
-
-    /* Contributor */
-
-    CCArray* contributors = CCArray::create();
-
-    ADD_PLAYER_NICK(contributors, "Awesomeoverkill", "ninXout", 7479054, 47290058, 77, 1, 5, -1);
-    ADD_PLAYER(contributors, "Capeling", 18226543, 160504868, 141, 76, 40, 40);
-    ADD_PLAYER(contributors, "TheSillyDoggo", 16778880, 162245660, 5, 41, 40, 12);
-    ADD_PLAYER_NICK(contributors, "angeld233", "angeld23", 28024715, 236381167, 98, 4, 12, 12);
-    ADD_PLAYER(contributors, "Uproxide", 25397826, 227796112, 296, 2, 12, 12);
-
-    auto* cellContrib = Build<GlobedCreditsCell>::create("Contributor", false, contributors)
-        .id("cell-contributor"_spr)
-        .parent(scrollLayer->m_contentLayer)
-        .collect();
-
-    /* Special */
-
-    CCArray* special = CCArray::create();
-
-    ADD_PLAYER(special, "MathieuAR", 3759035, 13781237, 220, 5, 9, 106);
-    ADD_PLAYER(special, "HJfod", 104257, 1817908, 132, 8, 3, 40);
-    ADD_PLAYER(special, "Cvolton", 761691, 6330800, 101, 19, 12, -1);
-    ADD_PLAYER(special, "alk1m123", 11535118, 116938760, 102, 12, 12, -1);
-    ADD_PLAYER(special, "mat4", 5568872, 19127913, 81, 0, 20, 2);
-
-    auto* cellSpecial = Build<GlobedCreditsCell>::create("Special thanks", true, special)
-        .id("cell-special"_spr)
-        .parent(scrollLayer->m_contentLayer)
-        .collect();
-
-    scrollLayer->m_contentLayer->setContentHeight(
-        cellOwners->getScaledContentSize().height
-        + cellStaff->getScaledContentSize().height
-        + cellContrib->getScaledContentSize().height
-        + cellSpecial->getScaledContentSize().height
-    );
-    scrollLayer->m_contentLayer->updateLayout();
-
-    // make the list start at the top instead of the bottom
-    util::ui::scrollToTop(scrollLayer);
+    /* Fetch credits from server */
+
+    web::AsyncWebRequest()
+        .fetch("http://127.0.0.1:8080/credits")
+        .json()
+        .then([scrollLayer](matjson::Value const& credits_data) {
```

Comment:
this will certainly not crash if we close the credits popup while it's loading

Reply:
ehhh